### DISCUSSION TITLE
[RFC] New Component App

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#38901] [RFCs - archive]
# 📋 RFC

## Summary

New App Component which provide global style & static function replacement.

## Basic example

```tsx
// Entry component
import { App } from 'antd';

export default ({ children }) => (
  <App>
    {children}
  </App>
);
```

```tsx
// Sub page
import { App } from 'antd';

export default () => {
  const { message } = App.useApp();

  React.useEffect(() => {
    message.success('Good!');
  }, []);

  return <div>Hello World</div>;
};
```

## Motivation

Static function in React 18 concurrent mode will not well support. In v5, we recommend to use hooks for the static replacement. But it will make user manual work on define this.

## Detailed

Support:
* v4 antd reset token version back (e.g. a link color)
* Static replacement of message.useMessage
* Static replacement of notification.useNotification
* Static replacement of Modal.useModal

## Drawbacks

Diff with static function, need a component wrapper.

## Adoption strategy

Apply new component in minor version.

**Comment by vagusX** [comment]
wow

**Comment by afc163** [comment]
```
<App resetCss={Boolean}>
```

是不是开一个字段可配。

**Reply by MadCcc** [reply]
reset css 不主动引入吧

**Reply by zombieJ** [reply]
全局的 reset 比如 html, body 还是用 `reset.css`。这个相当于 `ant-app` 里做了 a、input 之类的 reset。

**Comment by xiaohp** [comment]
可以换个名字么？App 一般用于项目入口组件，可能引起混淆

**Reply by zombieJ** [reply]
有更好的名字，欢迎提建议~

**Reply by TinsFox** [reply]
AntdProvide？

**Reply by kaoding** [reply]
Provide

**Reply by hec9527** [reply]
import { App as AntdApp } from 'antd';

**Reply by heiyu4585** [reply]
antdContainer

**Reply by Wxh16144** [reply]
`Driver` or `DriverGlobal` ?

**Comment by AshishKapoor** [comment]
Interesting 🤔

**Comment by phatify** [comment]
Why did Antd V5 decide to remove the base styles of global app and just focused on component styles?

Example: https://github.com/ant-design/ant-design/blob/4.x-stable/components/style/core/global.less

**Reply by afc163** [reply]
https://github.com/ant-design/ant-design/issues/9363

**Reply by phatify** [reply]
@afc163  looks like Antd v5 will be compatible when combined with more other libraries like TailwindCSS, WindiCSS, ... because Antd no longer depend on base styles global, right ?

**Comment by Huaxink** [comment]
UseApp is a poor method. If it is a public method, it cannot be used if you need to call static methods such as message.xxx

**Reply by Huaxink** [reply]
Since cssinjs has been used, why don't you get the configuration when calling static methods
